### DISCUSSION TITLE
Looking for some way to host nightly/latest (per commit) builds

### DISCUSSION THREAD

**Discussion by ericoporto** [community/community#77002] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi, I have a GitHub Action that builds a repository and I am currently pushing releases when a tag is created. These are the officially supported releases. For the purpose of testing things during the development process, by non-dev but still users of my software, I have been asked to make the builds that generated per build somewhere where the person could click and get the latest build. Because we would also to automatically download this latest build to run in other places, I would like to find somewhere to host this.

In the past I had used bintray for similar purposes but it's not available anymore.

I saw a way to force push as a single commit to a repository constantly [here](https://github.com/simonw/til/blob/1e29c3fe5e90c29b0e71d87dba805484ceb4393c/.github/workflows/build.yml#L80-L86), and in this way I could mark such repository as GitHub pages and have it as a nightly link. But I don't know if this is against the terms of service.

Is there any recommended way to do this? 

I don't need a history, I just need to have a fixed link that clicking always takes me to the latest result of the build. And I need to be alright to have people downloading from it.

**Comment by ChristopherHX** [comment, marked as answer]
I suggest to create a fixed tag like nightly and replace the asset of your release on every commit to your default branch. As far as I know this would have higher rate limites than using GitHub Pages.

For example [ghr](https://github.com/tcnksm/ghr) has an `-replace` flag to update the previous asset of the same name (delete and reupload).

> I saw a way to force push as a single commit to a repository constantly [here](https://github.com/simonw/til/blob/1e29c3fe5e90c29b0e71d87dba805484ceb4393c/.github/workflows/build.yml#L80-L86), and in this way I could mark such repository as GitHub pages and have it as a nightly link. But I don't know if this is against the terms of service.

I use GitHub Pages to host an apt and rpm nightly ppa, without need to forcepush using the new GitHub Actionbased deployment like [here](https://github.com/minecraft-linux/pkg/blob/af1cc4e317c7ee76aec27dbc41c2cc085cc7098c/.github/workflows/update-repo.yml).
Hosting an apt/rpm ppa is just impossible via GitHub Releases...
